armory/utils/metrics.py

```diff
@@ -8,8 +8,14 @@
 
 import logging
 import numpy as np
+import time
+from contextlib import contextmanager
+import cProfile
+import pstats
+import io
 
 logger = logging.getLogger(__name__)
+computational_resource_dict = {}
```

Comment:
I'm not a fan of having these module-level mutable containers. Can you look into making the computational_resource_dict a part of the MetricsLogger class and have it be passed in to the resource context?

Alternatively, the resource_context method could be a method of the MetricsLogger class.